datatypes/src/operations/image/colorizer.rs

```diff
@@ -669,6 +802,34 @@ mod tests {
         assert_eq!(color_table[2], RgbaColor::white());
     }
 
+    #[test]
+    fn logarithmic_color_table_with_extrema() {
+        let colorizer = Colorizer::logarithmic_gradient(
+            vec![
+                (5.0, RgbaColor::black()).try_into().unwrap(),
+                (15.0, RgbaColor::white()).try_into().unwrap(),
+            ],
+            RgbaColor::transparent(),
+            RgbaColor::pink(),
+            RgbaColor::pink(),
+        )
+        .unwrap();
+
+        let color_table = colorizer.color_table(5, 4., 16.);
+
+        assert_eq!(color_table.len(), 5);
+
+        // TODO: should it be like this here?
+        // or should the colorizer take the boundary points (5 & 15) and use the
+        // remaining slots to generate the table? In this case 5: black, 10: grey 15: white
+        // instead of evenly spacing it over the values including extrema?
+        assert_eq!(color_table[0], RgbaColor::pink()); // undercolor
+        assert_eq!(color_table[1], RgbaColor::new(78, 78, 78, 255)); // at 7
+        assert_eq!(color_table[2], RgbaColor::new(161, 161, 161, 255)); // at 10
+        assert_eq!(color_table[3], RgbaColor::new(222, 222, 222, 255)); // at 13
+        assert_eq!(color_table[4], RgbaColor::pink()); // overcolor
```

Comment:
If you specify a color for the value `5.0`, you want to have this color back and not another color.
So, the first should be black here and the last one white.

Reply:
I feel like I'm not understanding how breakpoints/colortables are supposed to work.
If the colorizer is specified with breakpoints `[5.,15.]` and we now generate a colortable of 5 values from `[4.,...,16.]`, shouldn't the first and last values be the over/under color in this case?

Reply:
Ah, I thought it was between 5 and 15. In this case, you are right.

You should set the over and under color to different values and then check that the implementation works.

Reply:
Changed with https://github.com/geo-engine/geoengine/commit/cacee41b1ac1cdda64e8404e7a84816f18cc8868